src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -35,6 +35,12 @@ internal partial class Binder
         /// <returns>True if a reference to "this" is available.</returns>
         internal bool HasThis(bool isExplicit, out bool inStaticContext)
         {
+            if (!isExplicit && IsInsideNameof && Compilation.IsFeatureEnabled(MessageID.IDS_FeatureInstanceMemberInNameof))
```

Comment:
I didn't follow why `!isExplicit` is needed in this check.

Reply:
My goal was to to use most conservative conditions to fix failing tests.